File: Data Structures & Algorithms/find-duplicate-integer/submission-4.java
class Solution {
    public int findDuplicate(int[] nums) {
        int slow = 0;
        int fast = 1;
        int size = nums.length;
        while (nums[slow] != nums[fast]) {
            slow = (slow + 1) % size;
            fast = (fast + 2) % size;

            if (slow == fast) {
                fast = (fast + 1) % size;
            }
        }
        return nums[slow];
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_nums = set(nums)
        maxi = 0
        for n in nums:
            increment = 0
            if (n-1) not in set_nums:
                while (n+increment) in set_nums:
                    increment += 1
                maxi = max(maxi, increment)

        return maxi 


"""

"""
        